brainglobe_atlasapi/atlas_generation/atlas_scripts/cavefish_atlas.py

```diff
@@ -0,0 +1,233 @@
+import os
+import time
+
+# from os import listdir, path
+# p = Path('.')
+from pathlib import Path
+
+import numpy as np
+import pooch
+import tifffile
+from rich.progress import track
+from scipy.ndimage import zoom
+
+from brainglobe_atlasapi import utils
+
+# from skimage import io
+from brainglobe_atlasapi.atlas_generation.mesh_utils import (
+    Region,
+    create_region_mesh,
+)
+from brainglobe_atlasapi.atlas_generation.wrapup import wrapup_atlas_from_data
+from brainglobe_atlasapi.structure_tree_util import get_structures_tree
+
+
+def create_atlas(working_dir, resolution):
+    ATLAS_NAME = "sju_cavefish"
+    SPECIES = "Astyanax mexicanus"
+    ATLAS_LINK = "https://a-cavefishneuroevoluti.vev.site/lab-website"
+    CITATION = "Kozol et al. 2023, https://doi.org/10.7554/eLife.80777"
+    ATLAS_FILE_URL = "https://cdn.vev.design/private/30dLuULhwBhk45Fm8dHoSpD6uG12/1hpojua-asty-atlas.zip"
+    ORIENTATION = "sla"
+    ROOT_ID = 999
+    ATLAS_PACKAGER = "Robert Kozol, kozolrobert@gmail.com"
+    ADDITIONAL_METADATA = {}
+
+    # setup folder for downloading
+
+    download_dir_path = working_dir / "downloads"
+    download_dir_path.mkdir(exist_ok=True)
+    atlas_path = download_dir_path / "atlas_files"
+
+    # download atlas files
+    utils.check_internet_connection()
+    pooch.retrieve(
+        ATLAS_FILE_URL,
+        known_hash="ee496465f1f55368aa1969180eff91863b800f57ea8dc5e9e3ab592fc480340e",
+        processor=pooch.Unzip(extract_dir=atlas_path),
+        progressbar=True,
+    )
+
+    structures_file = atlas_path / "asty_atlas/SPF2_25_Region_atlas_list.csv"
+    annotations_file = (
+        atlas_path / "asty_atlas/SPF2_regions_SP2c_1iWarp_25.tif"
+    )
+    reference_file = atlas_path / "asty_atlas/SPF2_terk_ref.tif"
+    meshes_dir_path = atlas_path / "asty_atlas/meshes"
+
+    # additional references (not in remote):
+    reference_cartpt = atlas_path / "asty_atlas/SPF2_cartpt_ref.tif"
+
+    try:
+        os.mkdir(meshes_dir_path)
+    except FileExistsError:
+        "mesh folder already exists"
+
+    # create dictionaries
+    print("Creating structure tree")
+    with open(
+        structures_file, mode="r", encoding="utf-8-sig"
+    ) as cavefish_file:
+        cavefish_dict_reader = csv.DictReader(cavefish_file)
+
+        # empty list to populate with dictionaries
+        hierarchy = []
+
+        # parse through csv file and populate hierarchy list
+        for row in cavefish_dict_reader:
+            hierarchy.append(row)
+
+    # make string to int and list of int conversions in
+    # 'id', 'structure_id_path', and 'rgb_triplet' key values
+    for i in range(0, len(hierarchy)):
+        hierarchy[i]["id"] = int(hierarchy[i]["id"])
+    for j in range(0, len(hierarchy)):
+        hierarchy[j]["structure_id_path"] = list(
+            map(int, hierarchy[j]["structure_id_path"].split("/"))
+        )
+    for k in range(0, len(hierarchy)):
+        try:
+            hierarchy[k]["rgb_triplet"] = list(
```

Comment:
I think this is copied from another script, but we don't need three identical loops?